### PR TITLE
feat(BUX-375): pulse simplification

### PR DESCRIPTION
**Pull request by wregulski**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1,135 +1,89 @@
...
-	Db            *Db               `mapstructure:"db"`
-	P2P           *p2pconfig.Config `mapstructure:"p2p"`
-	Merkleroot    *Merkleroot       `mapstructure:"merkleroot"`
-	Webhook       *Webhook          `mapstructure:"webhook"`
-	Websocket     *Websocket        `mapstructure:"websocket"`
-	HTTP          *HTTP             `mapstructure:"http"`
-	LoggerFactory logging.LoggerFactory
+// AppConfig returns strongly typed config values.
+type AppConfig struct {
+	DbConfig         *DbConfig         `mapstructure:"db"`
```
Comment (arkadiuszos4chain) [inline config/config.go:24]: Config.*Config? One "Config" isn't enough?

File: config/defaults.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		Type:               DBSqlite,
+		FilePath:           "./data/blockheaders.db",
+		Dsn:                "file:./data/blockheaders.db?_foreign_keys=true&pooling=true",
+		SchemaPath:         "./database/migrations",
+		PreparedDb:         false,
+		PreparedDbFilePath: "./data/blockheaders.csv.gz",
+	}
+}
+
+func getAuthConfigDefaults() *HTTPConfig {
```
Comment (arkadiuszos4chain) [inline config/defaults.go:35]: Name can be misleading. Maybe just getHttpConfigDefaults?

File: config.example.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  # Minimum sync peer network speed
+  min_sync_peer_network_speed: 51200
+  # Disable built-in checkpoints
+  disable_checkpoints: false
+  # Logging level for subsystems
+  log_level: info
+  # Maximum block size accepted by the node
+  excessive_block_size: 128000000
+  # Minimum time between attempts to send new inventory
+  trickle_interval: 50ms
```
Comment (dorzepowski) [inline config.example.yaml:35]: I'm not sure about exposing those configs (and by "not sure" I mean, I would suggest to not exposing them)

File: config.example.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+# P2P Configuration
+p2p:
+  # Maximum number of peers (inbound and outbound)
+  max_peers: 125
+  # Maximum number of peers per IP
+  max_peers_per_ip: 5
+  # Duration to ban misbehaving peers
+  ban_duration: 24h
+  # Minimum sync peer network speed
+  min_sync_peer_network_speed: 51200
```
Comment (dorzepowski) [inline config.example.yaml:27]: I'm not sure about exposing this config (and by "not sure" I mean, I would suggest to not exposing it)

File: config.example.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  prepared_db: false
+  # Path to prepared database file
+  prepared_db_file_path: "./data/blockheaders.csv.gz"
+
+# P2P Configuration
+p2p:
+  # Maximum number of peers (inbound and outbound)
+  max_peers: 125
+  # Maximum number of peers per IP
+  max_peers_per_ip: 5
```
Comment (dorzepowski) [inline config.example.yaml:23]: I'm not sure about exposing those configs (and by "not sure" I mean, I would suggest to not exposing them)

File: config/config.go (hunk 1 above)
Comment (wregulski) [reply config/config.go]: good point, removed redundant "Config" from struct names

File: config/defaults.go (hunk 2 above)
Comment (wregulski) [reply config/defaults.go]: done

File: internal/tests/testpulse/test_config-opts.go [hunk 6]
```diff
@@ -5,9 +5,9 @@ import (
 	"github.com/bitcoin-sv/pulse/internal/tests/testrepository"
 )
 
-// WithoutApiAuthorization disables authorization.
+// WithApiAuthorization allows to not use authorization in Pulse.
 func WithoutApiAuthorization() ConfigOpt {
```
Comment (dorzepowski) [inline internal/tests/testpulse/test_config-opts.go:9]: It is `Without` - means that authorization is disabled.
Maybe bettern name would be `WithApiAuthorizationDisabled`

File: transports/http/endpoints/api/access/access_endpoints_test.go [hunk 7]
```diff
@@ -46,39 +46,16 @@ func TestAccessEndpointWithWrongAuthHeader(t *testing.T) {
 	}
 }
 
-// Tests the GET /access endpoint with global auth token.
```
Comment (dorzepowski) [inline transports/http/endpoints/api/access/access_endpoints_test.go:49]: Why removed?

File: config.example.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+# HTTP Configuration
+http:
+  # Read timeout
+  read_timeout: 10
+  # Write timeout
+  write_timeout: 10
+  # HTTP server port
+  port: 8080
+  # URL prefix for API
+  url_prefix: "/api/v1"
```
Comment (dorzepowski) [inline config.example.yaml:67]: I would vote to not expose this config

**Review by arkadiuszos4chain — DISMISSED** [review]

File: transports/http/endpoints/api/access/access_endpoints_test.go (hunk 7 above)
Comment (wregulski) [reply transports/http/endpoints/api/access/access_endpoints_test.go:49]: Commented in a commit where this specific test was removed - I removed global auth token from config.

File: config.example.yaml (hunk 8 above)
Comment (wregulski) [reply config.example.yaml:67]: removed

File: internal/tests/testpulse/test_config-opts.go (hunk 6 above)
Comment (wregulski) [reply internal/tests/testpulse/test_config-opts.go:9]: It wasn't added by my, I see that my tool changed comment to different name. But anyway I switched to your recommendation:
**WithApiAuthorizationDisabled**

File: config.example.yaml (hunk 3 above)
Comment (wregulski) [reply config.example.yaml]: moved outside of the config, to a consts

File: config.example.yaml (hunk 4 above)
Comment (wregulski) [reply config.example.yaml]: moved outside of the config, to a consts

File: config.example.yaml (hunk 5 above)
Comment (wregulski) [reply config.example.yaml]: moved outside of the config, to a consts

File: transports/http/endpoints/api/access/access_endpoints_test.go (hunk 7 above)
Comment (wregulski) [reply transports/http/endpoints/api/access/access_endpoints_test.go:49]: Too long to explain :D had a different idea about auth token.

TL;DR:
Reverted this test.

**Review by dorzepowski — APPROVED** [review]
